tests/unit/s2n_connection_test.c

```diff
@@ -162,5 +163,88 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_connection_free(server_conn));
     }
 
+    /* Test: get selected digest alg */
+    {
+        struct s2n_connection *conn;
+        EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_CLIENT));
+
+        s2n_handshake_hash_algorithm output = { 0 };
+
+        EXPECT_FAILURE(s2n_connection_get_selected_digest_algorithm(NULL, &output));
+        EXPECT_FAILURE(s2n_connection_get_selected_digest_algorithm(conn, NULL));
+        EXPECT_FAILURE(s2n_connection_get_selected_client_cert_digest_algorithm(NULL, &output));
+        EXPECT_FAILURE(s2n_connection_get_selected_client_cert_digest_algorithm(conn, NULL));
+
+        EXPECT_SUCCESS(s2n_connection_get_selected_client_cert_digest_algorithm(conn, &output));
+        EXPECT_EQUAL(S2N_HANDSHAKE_HASH_NONE, output);
+
+        EXPECT_SUCCESS(s2n_connection_get_selected_digest_algorithm(conn, &output));
+        EXPECT_EQUAL(S2N_HANDSHAKE_HASH_NONE, output);
+
+        s2n_hash_algorithm inputs[] = { S2N_HASH_MD5, S2N_HASH_SHA1, S2N_HASH_SHA224, S2N_HASH_SHA256, S2N_HASH_SHA384, 
+                                        S2N_HASH_SHA512, S2N_HASH_SENTINEL, S2N_HASH_NONE };
+        EXPECT_EQUAL(S2N_HASH_SENTINEL, s2n_array_len(inputs));
```

Comment:
Since you're relying on order to map inputs -> expected_outputs, you could just not declare inputs and iterate over 0 to S2N_HASH_SENTINEL.

Reply:
I can do this for the Hash, but not the signature, since the signature enums have unique values. I can also make the signature sentinel equal to the number of signature enums